.github/workflows/sync-dependencies.yml

```
@@ -0,0 +1,48 @@
+  sync-dependencies:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+      - name: Sync template dependencies
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC1083:warning:26:4: This { is literal. Check expression (missing ;/\n?) or quote it

(shellcheck)